RadarSDK/RadarEvent+Internal.h

```diff
@@ -22,7 +22,7 @@
                      actualCreatedAt:(NSDate *_Nonnull)actualCreatedAt
                                 live:(BOOL)live
                                 type:(RadarEventType)type
-                          customType:(NSString *_Nullable)customType
+                      conversionType:(NSString *_Nullable)conversionType
```

Comment:
@david-goodfellow Do you foresee making this same change across the board in frontend, server, infra (kinesis, glue, athena, rollups, backfill, etc)?  Or leaving all of that as-is?

In my head I had sort of thought we would leave custom events as-in except for the syntactic sugar in the SDK, where we'd rename everything from "custom event" to "conversion"

Reply:
also, it looks like we wanted to change the `customType` param to `name` https://linear.app/radarlabs/issue/CORE-227/change-signature-of-custom-events-function-revenue-ios

should we keep this consistent between the external callsite and internal header?

Reply:
cc @joeychik-radar @jsani-radar

Reply:
the infra is going to stay the same but the logical lines in the sand I was making for namings was around repositories. E.g. in the sdk repos we should be consistent in using conversion everywhere

Reply:
per the `name` conversation, I thought this was just the name given in the ticket when Tim quickly wrote this up while I was OOO. @joeychik-radar and I had some discussions around this as shown in the image. We can go either way. 
<img width="504" alt="image" src="https://user-images.githubusercontent.com/61531269/223214500-7b229cdf-caf7-49fc-a4cd-0b3d65d7e8a1.png">

Reply:
per the naming for `name`/`customType`, I thought this was just the name given in the ticket when Tim quickly wrote this up while I was OOO. @joeychik-radar and I had some discussions around this as shown in the image. We can go either way.

Joey asking in our personal threads:
```
from the logConversion signature fun logConversion(name, metadata, callback), should we change name to type? since we pass the RadarEvent to the client, which doesn't have a name field
```

Reply:
`name` is more approachable than `type`, which seems like an internal impl detail

Reply:
should we rename the `customType` property on `RadarEvent` to `name` as well? or should we go for something more specific, as having a property called `name` seems a bit weird outside of the context of conversions

Reply:
my fault, for the internal `RadarEvent` model, having both `name` and `type` is inconsistent with the naming on the server and could be confusing

Reply:
`customType` here is syntactic sugar; events on the server don't have this field. The server passes down `type` as a string, and that gets stored as an enum in `RadarEvent`. The value of `customType` is whatever gets passed down from the server as `type` when `type` doesn't fit any of the enum values

Reply:
sgtm - since this is an internal impl detail, I'll defer to you all on the most intuitive / easiest to understand approach for future devs